### PR TITLE
[SE-4304] feat: add celerybeat configuration

### PR DESCRIPTION
**Pull request by gabor-boros**
This Pull Request adds `CELERYBEAT_SCHEDULER` to the configuration and sets it to its default value defined by Celery [documentation](https://docs.celeryproject.org/en/v4.4.7/userguide/configuration.html#beat-scheduler).

Defining the scheduler and adding the `--beat` flag to the workers allows scheduling tasks periodically.

**Dependencies**: None

**Sandbox URL**: N/A

**Merge deadline**: ASAP

**Testing instructions**:

* Proofread the configuration changes

**Author notes and concerns**:

This PR has a pair in the edx-platform repository: https://github.com/edx/edx-platform/pull/27630

**Reviewers**
- [ ] (OpenCraft internal reviewer's GitHub username goes here)
- [ ] edX reviewer[s] TBD

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @gabor-boros! I've created [OSPR-5783](https://openedx.atlassian.net/browse/OSPR-5783) to keep track of it in JIRA, where we prioritize reviews. Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


Please let us know once your PR is ready for our review and all tests are green.

**Comment by natabene** [comment]
@gabor-boros Thank you for your contribution. Please let me know once it is ready for our review.

**Comment by gabor-boros** [comment]
@natabene this is ready for edX review 🚀

**Comment by natabene** [comment]
@gabor-boros Could you please provide a user centric/business feature description  for this PR?

**Comment by adzuci** [comment]
@gabor-boros Thank you for submitting this PR.  I had a short discussion with the edX SRE team about this PR and am trying to understand the implications and long term impacts of adding a celery beat here.  edx.org does use celerybeat in one of our more intricate containerized services, but the engineering decisions around it's addition were not well documented and it was a bit of a best one of the not so great options.

Is OpenCraft currently using celerybeat to schedule jobs and if so could you explain what jobs and give a quick pro/con of this approach vs others?  For example, we could move away from using Jenkins jobs to schedule edx.org maintenance commands, but it's not clear to me if we would want to.

File: playbooks/roles/edxapp/templates/edx/app/supervisor/conf.d.available/workers.conf.j2 [hunk 1]
```diff
@@ -7,7 +7,7 @@ directory={{ edxapp_code_dir }}
 stdout_logfile={{ supervisor_log_dir }}/%(program_name)s-stdout.log
 stderr_logfile={{ supervisor_log_dir }}/%(program_name)s-stderr.log
 
-command={{ edxapp_app_dir }}/worker.sh --app={{ w.service_variant }}.celery:APP --loglevel=info --queues=edx.{{ w.service_variant }}.core.{{ w.queue }} --hostname=edx.{{ w.service_variant }}.core.{{ w.queue }}.%%h --concurrency={{ w.concurrency }} {{ '--maxtasksperchild ' + w.max_tasks_per_child|string if w.max_tasks_per_child is defined else '' }} {{ '--without-heartbeat' if not EDXAPP_CELERY_HEARTBEAT_ENABLED|bool else '' }} {{ '-O ' + w.prefetch_optimization if w.prefetch_optimization is defined else '' }}
+command={{ edxapp_app_dir }}/worker.sh --beat --app={{ w.service_variant }}.celery:APP --loglevel=info --queues=edx.{{ w.service_variant }}.core.{{ w.queue }} --hostname=edx.{{ w.service_variant }}.core.{{ w.queue }}.%%h --concurrency={{ w.concurrency }} {{ '--maxtasksperchild ' + w.max_tasks_per_child|string if w.max_tasks_per_child is defined else '' }} {{ '--without-heartbeat' if not EDXAPP_CELERY_HEARTBEAT_ENABLED|bool else '' }} {{ '-O ' + w.prefetch_optimization if w.prefetch_optimization is defined else '' }}
```
Comment (jdmulloy) [inline playbooks/roles/edxapp/templates/edx/app/supervisor/conf.d.available/workers.conf.j2]: I think we may prefer for there to be boolean variable to enable/disable the beats feature. 

```suggestion
command={{ edxapp_app_dir }}/worker.sh {{ '--beat' if EDXAPP_ENABLE_CELERY_BEAT|bool else '' }} --app={{ w.service_variant }}.celery:APP --loglevel=info --queues=edx.{{ w.service_variant }}.core.{{ w.queue }} --hostname=edx.{{ w.service_variant }}.core.{{ w.queue }}.%%h --concurrency={{ w.concurrency }} {{ '--maxtasksperchild ' + w.max_tasks_per_child|string if w.max_tasks_per_child is defined else '' }} {{ '--without-heartbeat' if not EDXAPP_CELERY_HEARTBEAT_ENABLED|bool else '' }} {{ '-O ' + w.prefetch_optimization if w.prefetch_optimization is defined else '' }}
```

File: playbooks/roles/edxapp/defaults/main.yml [hunk 2]
```diff
@@ -173,6 +173,7 @@ EDXAPP_CELERY_BROKER_VHOST: ""
 EDXAPP_CELERY_BROKER_USE_SSL: false
 EDXAPP_CELERY_EVENT_QUEUE_TTL: !!null
 EDXAPP_CELERY_TIMEZONE: "UTC"
+EDXAPP_CELERYBEAT_SCHEDULER: "celery.beat:PersistentScheduler"
```
Comment (jdmulloy) [inline playbooks/roles/edxapp/defaults/main.yml:176]:
```suggestion
EDXAPP_CELERYBEAT_SCHEDULER: "celery.beat:PersistentScheduler"
EDXAPP_ENABLE_CELERY_BEAT: false
```

**Review by jdmulloy — CHANGES_REQUESTED** [review]
I think we may prefer for this to be disabled by default with a boolean to enable it for those who want to use it.

**Comment by gabor-boros** [comment]
@jdmulloy Thank you for your suggestions! I just committed them.

@natabene, @adzuci, the reason we would need celery beat to run instructor reports (or other report generation, like the "learners" report from analytics) in a periodic way where we would set the frequency on the LMS Django Admin UI. For some of our clients, it is important to have the reports generated every X hours or days configured by them, but they don't want to expose the scheduling settings to their instructors or staff, but those who can access the Django Admin site.

Since this feature seems to be something that may not be needed for a broad range of edX users, we decided to create a [Django Plugin App](https://gitlab.com/opencraft/client/esme-learning/periodic-instructor-reports) to wrap Celery beat's periodic tasks and make it easier for our customers to configure the scheduling.

Although this kind of periodic task may not be used by a broad range of other edX users, we thought that it would be a good idea to let others have the possibility to enable Celery beat and set custom schedules, without forking the configuration repo. I think with the suggestions of @jdmulloy  this could be controlled in a more sophisticated way.

**Review by jdmulloy — APPROVED** [review]

**Comment by openedx-webhooks** [comment]
@gabor-boros 🎉 Your pull request was merged!

Please take a moment to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=edx/configuration&entry.867055334=https://github.com/edx/configuration/pull/6407&entry.1484655318=https://github.com/gabor-boros&entry.752974735=2021-05-16+08:24&entry.1917517419=2021-05-28+15:49&entry.2133058324=Yes) so we can improve your experience in the future.

**Pull request merged**